pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -1484,38 +1484,11 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 			var pciAddr string
 			pciAddr, sriovPciAddresses, err = popSRIOVPCIAddress(iface.Name, sriovPciAddresses)
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to configure SRIOV %s: %v", iface.Name, err)
 			}
-
-			dbsfFields, err := util.ParsePciAddress(pciAddr)
+			hostDev, err := createSRIOVHostDevice(pciAddr, iface.PciAddress, iface.BootOrder)
 			if err != nil {
-				return err
-			}
-
-			hostDev := HostDevice{
-				Source: HostDeviceSource{
-					Address: &Address{
-						Type:     "pci",
-						Domain:   "0x" + dbsfFields[0],
-						Bus:      "0x" + dbsfFields[1],
-						Slot:     "0x" + dbsfFields[2],
-						Function: "0x" + dbsfFields[3],
-					},
-				},
-				Type:    "pci",
-				Managed: "yes",
-			}
-
-			if iface.PciAddress != "" {
-				addr, err := decoratePciAddressField(iface.PciAddress)
-				if err != nil {
-					return fmt.Errorf("failed to configure SRIOV %s: %v", iface.Name, err)
-				}
-				hostDev.Address = addr
-			}
-
-			if iface.BootOrder != nil {
-				hostDev.BootOrder = &BootOrder{Order: *iface.BootOrder}
+				return fmt.Errorf("failed to configure SRIOV %s: %v", iface.Name, err)
```

Comment:
same

Reply:
In this case, the function `createSRIOVHostDevice` does not elaborate much about what did it tried to do that failed.
when there will be an error, it will not reflect in the logs that the this function failed due to hostDevice creation for SRIOV interface.

Reply:
It will say for example: `failed to configure SRIOV <iface-name>: failed to parse pci address <address>`.
What information is missing here?

Reply:
That it failed during hostDevice creation

Reply:
SRIOV is specifying the logical block in which it failed, which seems enough to me.
Based on what you expect here to have, each function used needs to add its name or what it does to the error chain. So a call stack of 20 will have 20 names chained. This is just a backtrace, and this is not how we log errors for sure.
(but if a projects decides that he wants such a thing, there are tooling to do this)